From f6681ddb41253f42b714a62fa3f1214acb8e2c7f Mon Sep 17 00:00:00 2001
From: Song Yoong Siang <yoong.siang.song@intel.com>
Date: Thu, 13 Jan 2022 14:46:58 +0800
Subject: [PATCH 27/42] net: stmmac: Disable MAC EEE when Network Proxy is
 enabled

Network Proxy is using PHY EEE Legacy mode without the need of MAC EEE.
Therefore, when Network Proxy is enabled, this commit disables MAC EEE
by setting dma_cap.eee to 0.

Note: It is unusual to change the value of any members in dma_cap.
A upstreamable design to prevent MAC EEE from running is needed.

Signed-off-by: Song Yoong Siang <yoong.siang.song@intel.com>
Signed-off-by: Noor Azura Ahmad Tarmizi <noor.azura.ahmad.tarmizi@intel.com>
---
 drivers/net/ethernet/stmicro/stmmac/stmmac_main.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/net/ethernet/stmicro/stmmac/stmmac_main.c b/drivers/net/ethernet/stmicro/stmmac/stmmac_main.c
index 62ee3d03c93c..1d25e27dba1d 100644
--- a/drivers/net/ethernet/stmicro/stmmac/stmmac_main.c
+++ b/drivers/net/ethernet/stmicro/stmmac/stmmac_main.c
@@ -7210,6 +7210,8 @@ int stmmac_dvr_probe(struct device *device,
 	if (priv->plat->has_netproxy) {
 		dev_info(priv->device, "Network Proxy supported\n");
 		device_set_wakeup_capable(priv->device, 1);
+		/* Network Proxy does not support MAC EEE */
+		priv->dma_cap.eee = 0;
 	}
 #endif
 
-- 
2.25.1

